src/Features/Core/Portable/Wrapping/InitializerExpression/InitializerExpressionCodeActionComputer.cs

```diff
@@ -0,0 +1,330 @@
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Wrapping.InitializerExpression
+{
+    internal abstract partial class AbstractInitializerExpression<TListSyntax, TListItemSyntax>
+    {
+        private class InitializerExpressionCodeActionComputer : AbstractCodeActionComputer<AbstractInitializerExpression<TListSyntax, TListItemSyntax>>
+        {
+            private readonly TListSyntax _listSyntax;
+            private readonly SeparatedSyntaxList<TListItemSyntax> _listItems;
+            private readonly SyntaxTrivia _afterOpenTokenIndentationTrivia;
+            private readonly SyntaxTrivia _singleIndentationTrivia;
+            private readonly SyntaxTrivia _braceIndentationTrivia;
+            private readonly bool _doWrapInitializerOpenBrace;
+
+            public InitializerExpressionCodeActionComputer(
+                AbstractInitializerExpression<TListSyntax, TListItemSyntax> service,
+                Document document, SourceText sourceText, DocumentOptionSet options,
+                TListSyntax listSyntax, SeparatedSyntaxList<TListItemSyntax> listItems,
+                bool doWrapInitializerOpenBrace, CancellationToken cancellationToken)
+                : base(service, document, sourceText, options, cancellationToken)
+            {
+                _listSyntax = listSyntax;
+                _listItems = listItems;
+
+                var generator = SyntaxGenerator.GetGenerator(this.OriginalDocument);
+
+                _afterOpenTokenIndentationTrivia = generator.Whitespace(GetAfterOpenTokenIdentation());
+                _singleIndentationTrivia = generator.Whitespace(GetSingleIdentation());
+                _braceIndentationTrivia = generator.Whitespace(GetBraceTokenIndentation());
+
+                _doWrapInitializerOpenBrace = doWrapInitializerOpenBrace;
+            }
+
+            private string GetAfterOpenTokenIdentation()
+            {
+                var openToken = _listSyntax.GetFirstToken();
+                var afterOpenTokenOffset = OriginalSourceText.GetOffset(openToken.Span.End);
+
+                var indentString = afterOpenTokenOffset.CreateIndentationString(UseTabs, TabSize);
+                return indentString;
+            }
+
+            private string GetBraceTokenIndentation()
+            {
+                var generator = SyntaxGenerator.GetGenerator(this.OriginalDocument);
+                var initialStatement = generator.GetDeclaration(_listSyntax);
+
+                if (initialStatement == null)
+                {
+                    return string.Empty;
+                }
+
+                var initialStatementOffset = initialStatement.SpanStart;
+                if (generator.GetDeclarationKind(initialStatement) == DeclarationKind.Method)
+                {
+                    // If we didn't find some form of declaration and are in a method body
+                    // the initializer must be in a return statement. Therefore, add a tab
```

Comment:
why does it matter if we're in a return statement or not?   if we care about being in a return statement, why not look for exactly that.

Reply:
When looking over declaration kind it appeared to find any kind an initializer could be used with besides a return statement. I figured if we ended up in a method without encountering a field, property, variable then it must be a return statement. However, it probably makes most sense to add Return as a DeclarationKind instead of making assumptions.